custom_model_runner/datarobot_drum/drum/language_predictors/r_predictor/score.R

```diff
@@ -321,4 +330,38 @@ predict_unstructured <- function(model=NULL, data, ...) {
     predictions <- do.call(score_unstructured_hook, kwargs_list)
     validated_pred_list = .validate_unstructured_predictions(predictions)
     validated_pred_list
+}
+
+#' Makes transforms against the model or by using the custom transform
+#' method and returns a list containing the transformed X and optionally y
+#'
+#'
+#' @param binary_data, Binary data containing X
+#' @param target_binary_data, Optional binary data containing y
+#' @param mimetype character, The file type of the binary data
+#' @param model to use to make predictions
+#'
+#' @return list, Two-element list containing transformed X (data.frame) and y (vector or NULL)
+#'
+outer_transform <- function(binary_data=NULL, target_binary_data=NULL, mimetype=NULL, model=NULL){
+    data <- .load_data(binary_data, mimetype=mimetype)
+    target_data <- NULL
+    if (!is.null(target_binary_data)) {
+        target_data <- .load_data(target_binary_data, use_hook=FALSE)
```

Comment:
why do we want to ignore the read hook for transform?

Reply:
that's only for reading in the target. The read hook is for X